### PR TITLE
feat: add CreateCanaryRelease method

### PR DESCRIPTION
**Pull request by LiteSun**

### CODE REVIEW

File: canary_release.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	// Value means the values that used in the eval
+	Value interface{} `json:"value,omitempty"`
+}
+
+// CanaryReleaseSpec is the specification of the CanaryRelease
+type CanaryReleaseSpec struct {
+	// Name specify the name of canary release
+	Name string `json:"name"`
+	// State specify the state of the canary release.Can be "pause", "in_progress", "finish"
+	State string `json:"state"`
```
Comment (tokers) [inline canary_release.go:55]: Should define constants for `pause`, `in_progress` and `finish`.

File: canary_release.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	// Status is status of canary release
+	Status EntityStatus `json:"status"`
+	// CreatedAt is the object creation time.
+	CreatedAt time.Time `json:"created_at"`
+	// UpdatedAt is the last modified time of this object.
+	UpdatedAt time.Time `json:"updated_at"`
+}
+
+// Rule is the rule that used in canary release
+type Rule struct {
```
Comment (tokers) [inline canary_release.go:39]: `Rule` is too generic. Let's add a prefix like `CanaryReleaseRule`.

File: canary_release.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	// Position means where we should get the key.Can be "header", "query" etc.
+	Position string `json:"position"`
+	// Key means the name of the key
+	Key string `json:"key"`
+	// Operator means the operator that used between key and value
+	Operator string `json:"operator"`
+	// Value means the values that used in the eval
```
Comment (tokers) [inline canary_release.go:46]:
```suggestion
	// Value means the values that used in the expression.
```

File: canary_release.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	// Percent specifies the percent of request will be transferred to canary upstream. Can be 0-100
+	Percent int `json:"percent,omitempty"`
+	// Rules specify the matched rules of request that should be transferred to canary upstream
+	Rules []Rule `json:"rules,omitempty"`
+}
+
+// CanaryReleaseInterface is the interface for manu
+type CanaryReleaseInterface interface {
+	// CreateCanaryRelease creates an API7 Cloud CanaryRelease in the specified Application.
+	// The given `cr` parameter should  specify the desired CanaryRelease specification.
```
Comment (tokers) [inline canary_release.go:69]:
```suggestion
	// The given `cr` parameter should specify the desired CanaryRelease specification.
```

File: common_types.go [hunk 5]
```diff
@@ -127,6 +127,7 @@ type ResourceCreateOptions struct {
 	// in the control plane. e.g., when creating Application, the
 	// ControlPlane.ID should be specified.
 	ControlPlane *ControlPlane
+	Application  *Application
```
Comment (tokers) [inline common_types.go:130]: Add more descriptions for this field.

**Review by nic-6443 — APPROVED** [review]

**Review by oil-oil — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**